src/lesson-4/src/actions/createTask.ts

```diff
@@ -0,0 +1,71 @@
+import { getFormHtml } from '../components/getFormHtml';
+import { createTask } from '../server-util';
+import type { Priority, Status, Task } from '../types';
+import { updateTasks } from './updateTasks';
+
+export const setupForm = async (element: HTMLElement) => {
+  element.innerHTML = getFormHtml();
+  const item = document.querySelector<HTMLFormElement>('#simpleForm')!;
+
+  const title = item.querySelector<HTMLInputElement>('#title')!;
+  const description = item.querySelector<HTMLInputElement>('#description')!;
+  const status = item.querySelector<HTMLSelectElement>('#status')!;
+  const priority = item.querySelector<HTMLSelectElement>('#priority')!;
+  const deadline = item.querySelector<HTMLInputElement>('#deadline')!;
+
+  const onSubmit = async (e: SubmitEvent) => {
+    e.preventDefault();
+
+    let valid = true;
+    const result: Omit<Task, 'id'> = {
+      title: title.value.trim(),
+      description: description.value.trim(),
+      createdAt: new Date(),
+      deadline: new Date(deadline.value),
+      status: status.value as Status,
+      priority: priority.value as Priority,
+    };
+
+    document.querySelectorAll('.error').forEach((e) => (e.textContent = ''));
+
+    if (result.title.length < 3) {
+      document.getElementById('titleError')!.textContent = 'Min. 3 symbols.';
```

Comment:
можна було б навіть окремий хелпер створити, щоб не повторювати `valid=false`, щось типу:
```suggestion
      setErrorText(errorId: string, text: string) { 
         document.getElementById(errorId)!.textContent = text;
         valid = false
      }
```